### DISCUSSION TITLE
LaserGRBL does not do 1 pass in Laser cut

### DISCUSSION THREAD

**Discussion by nissimzur** [arkypita/LaserGRBL#1490] [Q&A]
I create a cut for plastic sheet. See attached. 
The line is 0.3mm. 
1 pixel on Photoshop that was convert to SVG.
From unknown cause the LaserGRBL do 2 or more cut passes.  
Why? This is just double and triple the cut time and the cut is wider because of the temperature neer cut line. 
Thank you
Dr. Nissim Zur
[0.3 mm-cut.zip](https://github.com/arkypita/LaserGRBL/files/6419790/0.3.mm-cut.zip)

**Comment by gmmanonymus111** [comment]
I'm 99% certain that LaserGRBL is not at fault here. It is your method generating the SVG file, and it generates 2 lines very close to each other. Generating with InkScape, my SVG file is cut only once.
Side note: There is no need for line width setting when engraving SVG files. Line width equals to your laser point (or rectangle).
[tiger.zip](https://github.com/arkypita/LaserGRBL/files/6420475/tiger.zip)

**Comment by nissimzur** [comment]
Maybe  LaserGRBL is not at fault. But every SVG line has two sides. And LaserGRBL tries to cut both sides. No matter how the line is tiny.

**Comment by arkypita** [comment]
> ... But every SVG line has two sides...

This is not true. SVG lines are line and does not have sides.

**Comment by Armando578** [comment]
The answer is quite simple but took me a lot of time to think of it. The issue is, your draw cannot be hollow. Ill show you two images. The hollow one (second image) will be cut twice, as the program will interpret it as two different entities, but the filled one (first image) will be cut only once as the program sees it as only one entity. Hope that was helpful.
![Untitled](https://github.com/user-attachments/assets/47d5c9cb-13ba-44b1-a2ea-08e9ac3791ba)
![portalo d](https://github.com/user-attachments/assets/66dbad5f-172d-4f5c-9059-cff1d9fb5fec)

**Reply by Tribo-2dot0** [reply]
https://lasergrbl.com/wp-content/uploads/2022/08/g5785-1024x921.png

**Comment by LogrythmAtGmailDotCom** [comment]
There's also the case where two identical vectors ("lines") are perfectly stacked. 
In my case it's usually from me forgetting that I had duplicated a vector... 
Without deliberately checking, this is very hard to just see in an SVG editor and it only shows in LaserGBL as slightly thicker/brighter on screen.
